### PR TITLE
fix(#235): Error occurs when importing STD file during SCD automated …

### PR DESCRIPTION
**Pull request by SaintierFr**
…creation

Signed-off-by: SaintierFr <99645240+SaintierFr@users.noreply.github.com>

### CODE REVIEW

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java [hunk 1]
```diff
@@ -303,7 +304,7 @@ public static String stdCheckFormatExceptionMessage(TPrivate key) throws ScdExce
      * @param scdRootAdapter SCL file in which Private should be found
      * @return map of Private and its IEDName parameter
      */
-    public static Stream<TPrivate> createMapIEDNameAndPrivate(SclRootAdapter scdRootAdapter) {
+    public static Stream<TPrivate> getPrivatesStreamByIEDName(SclRootAdapter scdRootAdapter) {
```
Comment (samirromdhani) [inline sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java:307]: Here, `COMPAS_ICDHEADER` private was already set in the method filter, and javadoc indicates that it is an icdHeader Private not just Private. 
```suggestion
    public static Stream<TPrivate> getICDHeaderPrivatesStreamByIEDName(SclRootAdapter scdRootAdapter) {
```

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java [hunk 2]
```diff
@@ -312,7 +313,10 @@ public static Stream<TPrivate> createMapIEDNameAndPrivate(SclRootAdapter scdRoot
                 .filter(tPrivate ->
                         tPrivate.getType().equals(COMPAS_ICDHEADER.getPrivateType())
                                 && PrivateService.extractCompasICDHeader(tPrivate).isPresent()
-                                && PrivateService.extractCompasICDHeader(tPrivate).get().getIEDName() != null);
+                                && PrivateService.extractCompasICDHeader(tPrivate).get().getIEDName() != null)
+                .collect(Collectors.groupingBy(tPrivate -> PrivateService.extractCompasICDHeader(tPrivate).get().getIEDName()))
+                .values().stream()
+                .map(tPrivates -> tPrivates.get(0));
```
Comment (samirromdhani) [inline sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java:319]:
```suggestion
                .flatMap(Collection::stream)
```

**Review by samirromdhani — APPROVED** [review]

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java (hunk 1 above)
Comment (SaintierFr) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java:307]: Ok I renamed it and updated javadoc

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java [hunk 3]
```diff
@@ -312,7 +313,10 @@ public static Stream<TPrivate> createMapIEDNameAndPrivate(SclRootAdapter scdRoot
                 .filter(tPrivate ->
                         tPrivate.getType().equals(COMPAS_ICDHEADER.getPrivateType())
                                 && PrivateService.extractCompasICDHeader(tPrivate).isPresent()
-                                && PrivateService.extractCompasICDHeader(tPrivate).get().getIEDName() != null);
+                                && PrivateService.extractCompasICDHeader(tPrivate).get().getIEDName() != null)
```
[the remark this answers is not in the document]
Comment (SaintierFr) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java]: Ok

**Pull request merged**